### ISSUE TITLE
[electron][macOS] Attempting to call a function in a renderer window that has been closed or released

### ISSUE THREAD

**Issue by kittaakos** [eclipse-theia/theia#6923]
### Description
 

I can see the following warning quite often in the server log after opening multiple workspace roots from the electron application:
```
root WARN Attempting to call a function in a renderer window that has been closed or released.
Function provided here: 29.bundle.js:464:48
root WARN Attempting to call a function in a renderer window that has been closed or released.
Function provided here: 29.bundle.js:464:48
root WARN Attempting to call a function in a renderer window that has been closed or released.
Function provided here: 29.bundle.js:464:48
root WARN Attempting to call a function in a renderer window that has been closed or released.
Function provided here: 29.bundle.js:464:48
root WARN Attempting to call a function in a renderer window that has been closed or released.
Function provided here: 29.bundle.js:464:48
root WARN Attempting to call a function in a renderer window that has been closed or released.
Function provided here: 29.bundle.js:464:48
root WARN Attempting to call a function in a renderer window that has been closed or released.
Function provided here: 29.bundle.js:464:48
root WARN Attempting to call a function in a renderer window that has been closed or released.
Function provided here: 29.bundle.js:464:48
```

`29.bundle.js:464:48`:
```js
var ElectronMenuContribution = /** @class */ (function () {
    function ElectronMenuContribution(factory) {
        this.factory = factory;
    }
    ElectronMenuContribution.prototype.onStart = function (app) {
        var _this = this;
        this.hideTopPanel(app);
        this.setMenu();
        if (common_1.isOSX) {
            // OSX: Recreate the menus when changing windows.
            // OSX only has one menu bar for all windows, so we need to swap
            // between them as the user switches windows.
            electron.remote.getCurrentWindow().on('focus', function () { return _this.setMenu(); });
        }
```

Apparently, `electron.remote.getCurrentWindow()` is incorrect. One can reproduce it by having one or two windows with the electron example, close them so you have a single one, add and remove the focus on the window.

### Reproduction Steps

**OS and Theia version:**

**Diagnostics:**

**Comment by kittaakos** [comment]
Check if the windows are not GCd. We do not keep a ref to them. https://github.com/eclipse-theia/theia/pull/3556#issuecomment-440357911

**Comment by kittaakos** [comment]
Closed via #11075
